components/xx-rootmail.yaml

```diff
@@ -0,0 +1,329 @@
+AWSTemplateFormatVersion: 2010-09-09
+Transform: AWS::Serverless-2016-10-31
+
+Parameters:
+  Domain:
+    Type: String
+  Subdomain:
+    Default: aws
+    Type: String
+
+Mappings:
+  RegionMap: # https://docs.aws.amazon.com/general/latest/gr/s3.html#s3_website_region_endpoints
+    us-east-1:
+      Endpoint: s3-website-us-east-1.amazonaws.com
+      HostedZoneID: Z3AQBSTGFYJSTF
+    eu-central-1:
+      Endpoint: s3-website.eu-central-1.amazonaws.com
+      HostedZoneID: Z21DNDUVLTQW6Q
+    eu-west-1:
+      Endpoint: s3-website-eu-west-1.amazonaws.com
+      HostedZoneID: Z1BKCTXD74EZPE
+
+Outputs:
+
+  DelegationTarget:
+    Description: Nameservers for the hosted zone delegation
+    Value: !Join [ ',', !GetAtt HostedZone.NameServers ]
+
+Resources:
+
+  DomainBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Sub ${Subdomain}.${Domain}
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: true
+        BlockPublicPolicy: true
+        IgnorePublicAcls: true
+        RestrictPublicBuckets: true
+      WebsiteConfiguration:
+        RedirectAllRequestsTo:
+          HostName: "example.com"
+          Protocol: "https"
+
+  DomainBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref DomainBucket
+      PolicyDocument: !Sub |
```

Comment:
Nitpick: can be YAML as well, see e.g. https://github.com/superwerker/superwerker/blob/4d26141e3a4f325f0f5d742286bf560c14bde42c/components/00-control-tower.yaml#L151-L163

Reply:
I usually prefer JSON for policies - easier to generated / check with standard tools